pkg/render/logstorage/esgateway/esgateway.go

```diff
@@ -0,0 +1,313 @@
+package esgateway
+
+import (
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/render"
+	relasticsearch "github.com/tigera/operator/pkg/render/common/elasticsearch"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+)
+
+const (
+	DeploymentName        = "tigera-secure-es-gateway"
+	ServiceAccountName    = "tigera-secure-es-gateway"
+	RoleName              = "tigera-secure-es-gateway"
+	VolumeName            = "tigera-secure-es-gateway-certs"
+	ServiceName           = "tigera-secure-es-gateway-http"
+	ElasticsearchPortName = "es-gateway-elasticsearch-port"
+	KibanaPortName        = "es-gateway-kibana-port"
+	Port                  = 5554
+
+	ElasticsearchHTTPSEndpoint = "https://tigera-secure-es-http.tigera-elasticsearch.svc:9200"
+	ElasticsearchPort          = 9200
+	KibanaHTTPSEndpoint        = "https://tigera-secure-kb-http.tigera-kibana.svc:5601"
+	KibanaPort                 = 5601
+)
+
+func EsGateway(
+	installation *operatorv1.InstallationSpec,
+	pullSecrets []*corev1.Secret,
+	certSecrets []*corev1.Secret,
+	kubeControllersUserSecrets []*corev1.Secret,
+	kibanaInternalCertSecret *corev1.Secret,
+	esInternalCertSecret *corev1.Secret,
+	clusterDomain string,
+) render.Component {
+	var certSecretsESCopy []*corev1.Secret
+	secrets := certSecrets
+
+	// Copy the Operator namespaced cert secrets to the Elasticsearch namespace.
+	for _, s := range certSecrets {
+		certSecretsESCopy = append(certSecretsESCopy, secret.CopyToNamespace(render.ElasticsearchNamespace, s)...)
+	}
+	tlsAnnotations := map[string]string{render.ElasticsearchTLSHashAnnotation: rmeta.SecretsAnnotationHash(append(certSecretsESCopy, esInternalCertSecret)...)}
```

Comment:
Isn't esInternalCertSecret already in the namespace? If so, please don't re copy it back.

Reply:
It is but I'm not copying it here, just adding it to the tls annotation so that gateway is restarted if that secret changes.

Reply:
Oh right derp, sorry, didn't realise this was for the annotation.